torchgeo/datasets/geo.py

```diff
@@ -525,34 +532,89 @@ def __getitem__(self, query: GeoSlice) -> dict[str, Any]:
                 f'query: {query} not found in index with bounds: {self.bounds}'
             )
 
-        if self.separate_files:
-            data_list: list[Tensor] = []
-            filename_regex = re.compile(self.filename_regex, re.VERBOSE)
-            for band in self.bands:
-                band_filepaths = []
-                for filepath in index.filepath:
-                    filename = os.path.basename(filepath)
-                    directory = os.path.dirname(filepath)
-                    match = re.match(filename_regex, filename)
-                    if match:
-                        if 'band' in match.groupdict():
-                            start = match.start('band')
-                            end = match.end('band')
-                            filename = filename[:start] + band + filename[end:]
-                    filepath = os.path.join(directory, filename)
-                    band_filepaths.append(filepath)
-                data_list.append(self._merge_files(band_filepaths, query))
-            data = torch.cat(data_list)
+        if self.time_series:
+            # Group by unique datetime intervals to support timeseries
+            grouped_index = index.groupby(index.index)
+            time_steps = []
+            dates = []
+
+            for datetime_interval, group in grouped_index:
+                # Extract the center date of the interval
+                center_date = datetime_interval.mid
+                dates.append(center_date)
```

Comment:
I would like to make as many values to be Tensors as we can. In other places, we convert the datetime/pd.Timestamp object to a `.timestamp()` float that can be stored as a Tensor.

Reply:
Done